### DISCUSSION TITLE
TeslaMate companion App

### DISCUSSION THREAD

**Discussion by jnes92** [teslamate-org/teslamate#1575] [Show and tell]
Hey boys & girls, 

so I just got my new Tesla one month ago and Im using TeslaMate since the beginning.
Its really cool, but I wanted to get my data with me on the go. :) 

So I started developing a companion app, which connects to the TeslaMate DB and synchronizes the required data in a local database, so that I can see my latest drives even offline (my TeslaMate is running only in local area network, so no access from outside)

And now I just want to know, how you guys feel about this ? 
Would anybody else want to use this? 

Here is a snapshot (just started 1 week ago, so dont be brutal)
![UI-Snapshot-04_28](https://user-images.githubusercontent.com/10262902/116372060-59869c00-a80c-11eb-9516-a1d63466b6d8.jpeg)

Features so far:
- Sync Drives, Geofences, Addresses

**Comment by GtHbbrr** [comment]
Good idea! My teslamate runs in Google cloud and I would prefer an ios app that connects to the teslamate database and would function as Grafana in an app

**Reply by dhruvarora2** [reply]
Thats awesome! I would like to test it and hopefully learn about Swift development as a side project :p

**Reply by dhruvarora2** [reply]
@jnes92 How can I install it?

**Comment by bahree** [comment]
I am curious if there were any more updates on this? Is there a repro we can look at and try it out locally?

**Comment by djpalmis** [comment]
I'd really REALLY love a wearOs app for teslamate!

**Reply by darox** [reply]
I built an app for iOS https://fastcruise.app/

**Comment by jaypark0006** [comment]
This is really cool! But I’m wondering—does your iOS app connect directly to the database? I’m quite curious about how the data is being presented.

From my perspective, it might make more sense for the Teslamate server to act as an intermediary. In fact, what I mean is that we could eventually have multiple dashboards: for example, view (currently Grafana), manage/maintain (currently the web interface on port 4000), etc. That suggests we may need a dedicated server component to provide an HTTP API, which external apps like your iOS client could connect to directly.

**Comment by douwevt** [comment]
Now that sounds like a great idea! I would most definitely be interested! Thanks!

**Comment by mews-se** [comment]
Another one for this thread. I've spent the past few days building an iOS client for TeslaMate — it reads drives, charges and status from teslamateapi, and uses Grafana's datasource API for a map of everywhere the car has been.

It points at your own TeslaMate and nothing else: on the LAN at home, over VPN when you're away, or however you reach your instance. No account, no service in between, nothing leaving your own setup. And it would be free — there is nothing to sell here.

What's in it so far:

- Overview: battery, lock and sentry state, software version, location
- Drives grouped by day, each with the GPS track on a map and a speed graph you can drag along to follow the car through the trip
- Charges with AC/DC split, the charge curve, and cost per session
- Statistics per week, month and year — distance, energy, cost, consumption
- Optional: a Tessie key fills in the Supercharger costs TeslaMate never gets from Tesla
- English and Swedish, light and dark

<p align="center">
  <img width="250" alt="Overview" src="https://github.com/user-attachments/assets/b5804aaa-00c6-4e33-8827-21ca8fb5977f" />
  <img width="250" alt="Drive detail" src="https://github.com/user-attachments/assets/0c847dcf-afa1-4d40-ae04-90f880733167" />
  <img width="250" alt="Charge detail" src="https://github.com/user-attachments/assets/329dd5f2-c061-4934-b884-5bbfa881be11" />
</p>
<p align="center">
  <img width="250" alt="Charges" src="https://github.com/user-attachments/assets/c7355e72-9cd2-48c5-910f-d9f66da70441" />
  <img width="250" alt="Statistics" src="https://github.com/user-attachments/assets/8c0dd7c0-15a8-407d-862f-64aa5682609b" />
</p>

It runs on my own phone with a free provisioning profile today, which means reinstalling it every seven days. Putting it on the App Store costs $99 a year and I don't have an Apple developer account any more, so before spending that I'd like to know whether anyone else would actually use it.

The source isn't public yet, but it will most likely end up on GitHub, so anyone who wants to can file issues or send pull requests.

It's an unofficial community tool, not affiliated with, endorsed by or supported by the TeslaMate project.

**Reply by mews-se** [reply]
Actually I decided to pay up anyways :) The project will be open source and version 1 is in Apple review

https://github.com/mews-se/evlog-ios

**Reply by mews-se** [reply]
It's live finally - and v2 is more or less done!
